proxy/src/control/destination/background.rs

```diff
@@ -211,7 +222,7 @@ where
                             if set.query.is_none() {
                                 set.reset_dns_query(
                                     &self.dns_resolver,
-                                    Duration::from_secs(0),
+                                    Instant::now(),
```

Comment:
In general, when should we use `Instant::now()` and when should we use Tokio's `now()`?

Reply:
When I wrote this, I was still waiting for a resolution to #973, and it wasn't clear what should be used. Now, it seems like Tokio's `now` would be correct, but there's currently no way for us to get the `Now` instance here. There's been some discussion of adding support for that to Tokio, at which point we can replace uses of `Instant::now()` throughout the proxy.